tests/test_frontdoor.py

```diff
@@ -304,6 +304,41 @@ def test_back_button():
     assert form["country"] == "England"
 
 
+@pytest.mark.parametrize(
+    ("supplier_name", "expected_text"),
+    [
+        ("Shell", "Shell is now owned by Octopus Energy."),
+        ("Bulb, now part of Octopus Energy", "Bulb is now owned by Octopus Energy."),
+        ("Utility Warehouse", "Referral requests from UW customers will be managed by E.ON Next"),
+    ],
+)
+def test_own_property_back_button_with_supplier_should_return_to_supplier_warning_page(supplier_name, expected_text):
```

Comment:
Are we testing the happy path somewhere else?

Reply:
What is the happy path? @jdgage

Reply:
I've looked into it, and it seems test_back_button might fit the bill?
```
    form = page.get_form()
    assert form["own_property"] == "Yes, I own my property and live in it"

    page = page.click(contains="Back")

    form = page.get_form()
    assert form["supplier"] == "Utilita"

    page = page.click(contains="Back")

    form = page.get_form()
    assert form["country"] == "England"
```

Reply:
Happy path in this case is that the back button does what we expect when there's not a warning page. So yes, it looks like that test covers it.

In general, it's context dependent, but it's usually going to be the journey the user takes through a system when everything's 'normal', there aren't errors, warnings, etc. Most of the time it will be the most common path through the system, but that's not necessarily true.